src/paths.ts

```diff
@@ -6,34 +6,95 @@ import * as path from 'path';
 import {decode} from 'urlencode';
 
 /** converts a server-side XDebug file URI to a local path for VS Code with respect to source root settings */
-export function convertDebuggerPathToClient(fileUri: string|url.Url, localSourceRoot?: string, serverSourceRoot?: string): string {
+export function convertDebuggerPathToClient(fileUri: string|url.Url, localSourceRoot?: string, serverSourceRoot?: string, pathMapping?: { [index: string]: string; }): string {
```

Comment:
Users that update will use "localSourceRoot" and "serverSourceRoot". It was easier to be backwards compatible with an optional extra argument, but I can certainly reformat it to inject localSourceRoot and serverSourceRoot into the config object if they exist.

(this is my first open source node contribution so I apologize for any code style differences)